airbyte-webapp/src/components/ui/ResizablePanels/ResizablePanels.tsx

```diff
@@ -0,0 +1,107 @@
+import { faGripLinesVertical } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import classNames from "classnames";
+import React from "react";
+import { ReflexContainer, ReflexElement, ReflexSplitter } from "react-reflex";
+import { useWindowSize } from "react-use";
+
+import styles from "./ResizablePanels.module.scss";
+
+interface ResizablePanelsProps {
+  leftPanel: {
+    children: React.ReactNode;
+    smallWidthHeader: React.ReactNode;
+    className?: string;
+  };
+  rightPanel: {
+    children: React.ReactNode;
+    smallWidthHeader: React.ReactNode;
+    showPanel: boolean;
+    className?: string;
+    startingFlex?: number;
+  };
+  containerClassName?: string;
+}
```

Comment:
So I tried out the approach you suggested above, since I agree that would be better based on the reasons you listed above.

Unfortunately, the Reflex library seems to _really_ not like it when I try to wrap the child components (ReflexElement, ReflexSplitter) in a separate component, instantiate them elsewhere, and pass them in. 
First it threw errors saying that I couldn't pass refs into functional components. So I tried switching them to class-based components, which allowed me to render the page, but the resizing functionality was still completely broken (throwing a bunch of unclear errors whenever I tried to resize).

I pushed those changes up to a branch here if you want to try playing around with it at all yourself: https://github.com/airbytehq/airbyte/tree/lmossman/convert-resizable-panels-to-library

But it seems like this approach is just not very compatible with the library we are using. I didn't want to spend too much time on this, so I opted to instead just simplify the ResizablePanels class a bit more based on your other suggestions, and keep the more complex props for now. Let me know if you don't agree with that choice!

Reply:
Ah yes that is a bit tricky with the refs. Thanks for throwing up the branch - I [forked it](https://github.com/airbytehq/airbyte/compare/joey/convert-resizable-panels-to-library) and made a couple changes here: https://github.com/airbytehq/airbyte/blob/joey/convert-resizable-panels-to-library/airbyte-webapp/src/components/ui/ResizablePanels/ResizablePanels.tsx

I think it should work as expected, you just have to explicitly forward the ref (`React.forwardRef`) and also spread props in a couple places.

Anyway, fine with me to go ahead with either approach!

Reply:
Thank you for looking into this and linking to the documentation! I will read up on forwarding refs since my knowledge was definitely lacking in that area

Reply:
I tried applying the changes in your branch, and they did fix the errors and made the resizing functionality work.

However, it now introduces this problem, where when I initialize something like
```
<ResizablePanels>
  <Panel>
    {content}
  </Panel>
  <Splitter />
  <Panel>
    {content}
  </Panel>
</ResizablePanels>
``` 
it seems that the ReflexContainer is seeing three children elements passed in, so it reserves space for 3 separate panels (i.e. it only calculates the flex-grow values for the panels to make them take up 2/3 of the space, rather than to take up the whole space), so we end up with a page that looks like this:
<img width="1968" alt="Screen Shot 2022-10-24 at 2 56 19 PM" src="https://user-images.githubusercontent.com/22731524/197637939-589d47fa-bd14-4d1b-92b7-e05c6782983a.png">

It seems like the ReflexContainer needs to have the ReflexSplitter be initialized directly as a child for it to know to not reserve an entire panels worth of space for it. I tried playing around with this more to see if I could get it to work but I couldn't figure out a fix for this. I pushed up the most recent changes for that approach here: https://github.com/airbytehq/airbyte/tree/lmossman/resizable-panels-library-with-refs

So I think I'll just need to merge this PR with the current approach in order to unblock myself here. Thanks for the suggestions though, I learned a lot!